### PR TITLE
DAOS-13085: update libfabric to 1.18.0

### PR DESCRIPTION
**Pull request by soumagne**
Apply also patch for prov/verbs QP error state recovery (DAOS-12991)

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by frostedcmos — DISMISSED** [review]

**Review by jolivier23 — DISMISSED** [review]

File: debian/control [hunk 1]
```diff
@@ -171,7 +171,7 @@ Section: net
 Architecture: any
 Multi-Arch: same
 Depends: ${shlibs:Depends}, ${misc:Depends}, openmpi-bin,
-         ipmctl (>=03.00.00.0468), libfabric (>= 1.15.1-1), libfabric (<< 1.18), spdk-tools (>= 22.01.2)
+         ipmctl (>=03.00.00.0468), libfabric (>= 1.15.1-1), spdk-tools (>= 22.01.2)
```
Comment (brianjmurrell) [inline debian/control:174]: Ditto.

File: debian/control [hunk 2]
```diff
@@ -152,7 +152,7 @@ Package: daos-client
 Section: net
 Architecture: any
 Multi-Arch: same
-Depends: ${shlibs:Depends}, ${misc:Depends}, openmpi-bin, libfabric (>= 1.15.1-1), libfabric (<< 1.18)
+Depends: ${shlibs:Depends}, ${misc:Depends}, openmpi-bin, libfabric (>= 1.15.1-1)
```
Comment (brianjmurrell) [inline debian/control:155]: I think it's actually the library you want to pin here:
```suggestion
Depends: ${shlibs:Depends}, ${misc:Depends}, openmpi-bin, libfabric1 (>= 1.15.1-1)
```
or are there userspace tools that daos-server needs?

File: ci/provisioning/post_provision_config_common_functions.sh [hunk 3]
```diff
@@ -259,6 +259,25 @@ post_provision_config_nodes() {
...
+                    build_number="${branch#*:}"
+                    branch="${branch%:*}"
+                fi
+            fi
+            local repo_url="${JENKINS_URL}"job/daos-stack/job/"${repo}"/job/"${branch//\//%252F}"/"${build_number}"/artifact/artifacts/$DISTRO_NAME/
+            dnf -y config-manager --add-repo="${repo_url}"
+            disable_gpg_check "$repo_url"
+        done
+    fi
+
```
Comment (brianjmurrell) [inline ci/provisioning/post_provision_config_common_functions.sh:280]: I'm not sure we want to land this without some more thought on potential fall-out and testing.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: debian/control (hunk 2 above)
Comment (soumagne) [reply debian/control:155]: yeah CI relies on the fi_info tools for now. In the near future I'm planning to remove the entire dependency on libfabric from DAOS server so I would not read too much into it at this point.

Comment (brianjmurrell) [reply debian/control:155]: Is it just CI that relies on those tools or does the DAOS product rely on them also?

**Review by frostedcmos — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

File: debian/control (hunk 2 above)
Comment (brianjmurrell) [reply debian/control:155]: So this really should be a `Depends:` on a daos*tests* package, yes?

Comment (soumagne) [reply debian/control:155]: right, if we wanted to do that, yes. I will make a PR relatively soon that removes the direct dependency on libfabric (and UCX) from DAOS (by making the control plane no longer directly query from them) so I would expect a lot more of simplification at that time.

**Pull request merged**